### PR TITLE
Rework of Spotify Synchronization

### PR DESCRIPTION
**Pull request by DanielW1987**
The next step for Spotify Synchronization.

- [x] Create own `SpotifyAuthorizationRestController`
- [x] Spotify Synchronization UI
- [x] Connect with Spotify
- [x] Fetch artists
- [ ] Select artists for synchronization
- [ ] Submit selected artists 
- [ ] Naming issues (import vs. synchronize)

Please test it on your local machine. Maybe you have to completely remove your local database.

### CODE REVIEW

File: webapp/src/main/resources/static/ts/src/service/spotify-synchronization-render-service.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        authorizationResponse.then(response => window.location.href = response.authorizationUrl);
+    }
+
+    private synchronizeSpotifyArtists(): void {
+        // ToDo DanielW:
+        //  - show loading indicator
+        //  - prevent multiple fetches
+        const followedArtists = this.spotifyRestClient.fetchFollowedArtists();
+        followedArtists.then(response => {
+            console.log(response.artists);
```
Comment (hedgehog1833) [inline webapp/src/main/resources/static/ts/src/service/spotify-synchronization-render-service.ts:84]:
```suggestion
```

File: webapp/src/main/resources/templates/frontend/spotify-synchronization.html [hunk 2]
```diff
@@ -5,16 +5,115 @@
...
                 <h2 class="h5 mt-3">Spotify Synchronization</h2>
                 <hr class="border-secondary"/>
-                <button type="button" class="btn btn-success" id="authorization-button">Authorize Spotify</button>
+
+                <div id="button-bar"></div>
+                <div class="mt-3 text-right invisible" id="foo">
+                    <a class="light-hyperlink mr-2" href="#">Select all</a> | <a class="light-hyperlink ml-2" href="#">Deselect all</a>
+                </div>
+                <div id="artists-container">
+<!--                    <div class="card mt-2 spotify-synchro-card pointer">-->
```
Comment (hedgehog1833) [inline webapp/src/main/resources/templates/frontend/spotify-synchronization.html:40]: All this commented stuff can go right?

File: webapp/src/test/java/rocks/metaldetector/service/spotify/SpotifyUserAuthorizationServiceImplTest.java [hunk 3]
```diff
@@ -40,27 +42,95 @@
...
+      // when
+      boolean result = underTest.exists();
+
+      // then
+      assertThat(result).isTrue();
+    }
+
+    @ParameterizedTest(name = "should return false for <{0}>")
+    @MethodSource("incompleteAuthorizationEntityProvider")
+    @DisplayName("should return false if access token and refresh token not exist")
```
Comment (hedgehog1833) [inline webapp/src/test/java/rocks/metaldetector/service/spotify/SpotifyUserAuthorizationServiceImplTest.java:114]:
```suggestion
    @DisplayName("should return false if access token or refresh token don't exist")
```

File: webapp/src/main/resources/templates/frontend/spotify-synchronization.html (hunk 2 above)
Comment (DanielW1987) [reply webapp/src/main/resources/templates/frontend/spotify-synchronization.html:40]: not yet, I still need parts of it as templates

**Review by hedgehog1833 — APPROVED** [review]

**Comment by DanielW1987** [comment]
[Frontend to synchronize artists from Spotify](https://trello.com/c/q4gIirGx/247-frontend-to-synchronize-artists-from-spotify)

**Pull request merged**